src/test/AutoRoller.t.sol

```diff
@@ -936,12 +978,195 @@ contract AutoRollerTest is DSTestPlus {
         assertEq(targetedRate, 0);
+
+        vm.warp(maturity + autoRoller.cooldown());
+
+        // Roll into new series
+        autoRoller.roll();
+
+        // Scale goes up
+        mockAdapter.setScale(1.9e18);
+        
+        maturity = autoRoller.maturity();
+        vm.warp(maturity);
+
+        autoRoller.settle();
+
+        // Targeted rate is not 0 TODO: maybe assert against the rate that should be 
+        targetedRate = utils.getNewTargetedRate(0, address(mockAdapter), maturity, space);
+        assertGt(targetedRate, 0);
+    }
+
+    // OTHER TESTS (LINES NOT COVERED ACCORDING CODECOV) TODO: re-order these tests?
+
+    function testAfterDepositWithOnlyTargetLiquidity() public {
+        // The idea is making a test that covers line case on the `afterDeposit` function
+        // where `if (assets - targetToJoin > 0)` is false.
+        // I think can never happen since `onSponsorWindowOpened`, when calling `_space.getEQReserves`
+        // we are doing `targetedRate < 0.01e18 ? 0.01e18 : targetedRate` which means that we are always
+        // getting a `eqPTReserves` > 0 so we will be issuing PTs. 
+    }
+
+    function testPreviewWithdrawInsufficentLiquidity(uint256 amount) public {
+        // 1. Roll into the first Series.
+        autoRoller.roll();
+
+        // 2. Make a deposit.
+        autoRoller.deposit(1e18, address(this));
+
+        // 3. Expect revert because of InsufficientLiquidity.
+        vm.expectRevert(abi.encodeWithSelector(AutoRoller.InsufficientLiquidity.selector));
+        autoRoller.previewWithdraw(100e18);
+    }
+
+    function testMaxWithdrawDuringCooldown(uint256 amount) public {
+        // 1. Make a deposit durring cooldown.
+        autoRoller.deposit(1e18, address(this));
+
+        // 2. Assert max withdraw is same as deposit.
+        uint256 maxWithdraw = autoRoller.maxWithdraw(address(this));
+        assertEq(maxWithdraw, 1e18);
+    }
+
+    function testRedeemWithYTsExcess() public {
+        // 1. Deposit during the initial cooldown phase.
+        autoRoller.deposit(0.2e18, address(this));
+        assertEq(autoRoller.balanceOf(address(this)), 0.2e18);
+
+        // 3. Roll the Target into the first Series.
+        autoRoller.roll();
+
+        // 4. Deposit during the first active phase.
+        autoRoller.deposit(0.3e18, address(this));
+        assertRelApproxEq(autoRoller.balanceOf(address(this)), 0.5e18, 0.0001e18 /* 0.01% */);
+
+        // 5. Redeem all shares while still in the active phase.
+        uint256 targetBalPre = target.balanceOf(address(this));
+        vm.expectCall(address(periphery), abi.encodeWithSelector(periphery.swapYTsForTarget.selector));
+        autoRoller.redeem(autoRoller.balanceOf(address(this)), address(this), address(this));
+        uint256 targetBalPost = target.balanceOf(address(this));
+        // assertRelApproxEq(targetBalPost - targetBalPre, 0.5e18, 0.0001e18 /* 0.01% */); // TODO: what should it be the expected value?
```

Comment:
I think we should expect it to be pretty close to "total deposited amount - `firstDeposit`" (since `firstDeposit` is kept locked in the contract forever, similar to the min bpt for our space pool). What were you seeing here for this val?

Reply:
hmm yeah, i was seeing something similar (I forgot to consider the `firstDeposit`) but now, considering the `firstDeposit`:
- I have deposited in total `0.5e18` and I'm asserting that `0.5e18 - 0.01e18` (total deposited - `firstDeposit`) should almost equal to the target I received back which is `499642339889372330`, shouldn't that be lower? Or how do I establish which is a good delta in this case?